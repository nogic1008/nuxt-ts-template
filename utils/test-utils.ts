/** Generate [0-9a-z]{length} string. */
export const generateRandomString = (length: number) =>
  [...Array(Number.isInteger(length) ? length : 12)]
    .map(() => (~~(Math.random() * 36)).toString(36))
    .join('')
