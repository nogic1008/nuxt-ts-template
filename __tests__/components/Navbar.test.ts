import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils'
import Buefy from 'buefy'
import { IVueI18n } from 'vue-i18n'

import NavbarComponent from '~/components/Navbar.vue'

const localVue = createLocalVue()
localVue.use(Buefy)
localVue.mixin({
  methods: {
    localePath: (obj: object) => obj,
    switchLocalePath: (code: string) => code
  }
})

describe('components/Navbar.vue', () => {
  const $i18n: Partial<IVueI18n> = {
    locale: 'en',
    locales: [
      { code: 'en', iso: 'en-US', flag: 'us', name: 'English' },
      { code: 'ja', iso: 'ja-JP', flag: 'jp', name: '日本語' }
    ]
  }

  test('renders correctly', () => {
    const wrapper = mount(NavbarComponent, {
      localVue,
      mocks: { $i18n },
      stubs: {
        NuxtLink: RouterLinkStub,
        Nuxt: true
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
