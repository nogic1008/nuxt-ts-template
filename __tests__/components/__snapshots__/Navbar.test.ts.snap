// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navbar.vue { isOpened: false } renders correctly 1`] = `
<nav
  aria-label="main navigation"
  class="navbar is-primary"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <a
      class="navbar-item"
    >
      <img
        alt="Buefy"
        height="28"
        src="~assets/buefy.png"
      />
    </a>
     
    <a
      aria-label="menu"
      class="navbar-burger burger"
      role="button"
      tabindex="0"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
   
  <div
    class="navbar-menu"
  >
    <div
      class="navbar-start"
    />
     
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <span
          aria-haspopup="true"
          class="navbar-link"
          tabindex="0"
        >
          <flag-stub
            iso="us"
          />
           English
        
        </span>
         
        <div
          class="navbar-dropdown is-right"
        >
          <a
            class="navbar-item"
          >
            <flag-stub
              iso="jp"
            />
             
            <span>
               日本語
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</nav>
`;

exports[`components/Navbar.vue { isOpened: true } renders correctly 1`] = `
<nav
  aria-label="main navigation"
  class="navbar is-primary"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <a
      class="navbar-item"
    >
      <img
        alt="Buefy"
        height="28"
        src="~assets/buefy.png"
      />
    </a>
     
    <a
      aria-label="menu"
      class="navbar-burger burger"
      role="button"
      tabindex="0"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
   
  <div
    class="navbar-menu is-active"
  >
    <div
      class="navbar-start"
    />
     
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item has-dropdown is-hoverable"
      >
        <span
          aria-haspopup="true"
          class="navbar-link"
          tabindex="0"
        >
          <flag-stub
            iso="us"
          />
           English
        
        </span>
         
        <div
          class="navbar-dropdown is-right"
        >
          <a
            class="navbar-item"
          >
            <flag-stub
              iso="jp"
            />
             
            <span>
               日本語
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
